pkg/routeagent_driver/handlers/ovn/gateway_route_handler.go

```diff
@@ -68,20 +68,28 @@ func (h *GatewayRouteHandler) GetNetworkPlugins() []string {
 
 func (h *GatewayRouteHandler) RemoteEndpointCreated(endpoint *submarinerv1.Endpoint) error {
 	if h.State().IsOnGateway() {
-		_, err := h.smClient.SubmarinerV1().GatewayRoutes(endpoint.Namespace).Create(context.TODO(),
-			h.newGatewayRoute(endpoint), metav1.CreateOptions{})
-		if err != nil && !apierrors.IsAlreadyExists(err) {
+		logger.V(log.TRACE).Infof("Remote endpoint %q created event received on gateway node", endpoint.Name)
+
+		gwr := h.newGatewayRoute(endpoint)
+
+		result, err := util.CreateOrUpdate(context.TODO(), GatewayResourceInterface(h.smClient, endpoint.Namespace),
+			gwr, util.Replace(gwr))
+		if err != nil {
 			return errors.Wrapf(err, "error processing the remote endpoint creation for %q", endpoint.Name)
 		}
+
+		logger.V(log.TRACE).Infof("GatewayRoute %s: %#v", result, gwr)
 	}
 
 	return nil
 }
 
 func (h *GatewayRouteHandler) RemoteEndpointRemoved(endpoint *submarinerv1.Endpoint) error {
 	if h.State().IsOnGateway() {
+		logger.V(log.TRACE).Infof("Endpoint %q removed event received on gateway node ", endpoint.Name)
+
 		if err := h.smClient.SubmarinerV1().GatewayRoutes(endpoint.Namespace).Delete(context.TODO(),
-			endpoint.Name, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
+			endpoint.Spec.ClusterID, metav1.DeleteOptions{}); err != nil && !apierrors.IsNotFound(err) {
```

Comment:
If we start using `ClusterID` instead of `Name`, will there be any impact during upgrades where the old code might have created/programmed rules based on endpoint.Name. Will they remain as stale and create issues?

Reply:
good catch. added a cleanup code in tranistiontogateway()